core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -375,9 +376,13 @@ public boolean hasLostStorage() {
 
   @Override
   public String toString() {
+    int blockSizeLimit = 100;
     return MoreObjects.toStringHelper(this).add("id", mId).add("workerAddress", mWorkerAddress)
         .add("capacityBytes", mCapacityBytes).add("usedBytes", mUsedBytes)
-        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs).add("blocks", mBlocks)
+        .add("lastUpdatedTimeMs", mLastUpdatedTimeMs)
+        .add("blocks",
+              (mBlocks.size() < blockSizeLimit) ? mBlocks :
```

Comment:
you shouldn't need this condition with `limit`.

Reply:
I think adding this condition will reduce one copy if we don't need to truncate?